[Jim Vincent] iP

jimvae opened the pull request.

Commit:
Add Gradle support

Commit:
Add Chatbot.java

Commit:
Chatbot class:
*Add a planner (list of string)
*Add method to add events to the planner
*Add method to print out the events inside the planner
*Modify the chat method to add what the user input and add a case for the input "list"

Commit:
Add Task.java and Update Chatbot

Revised Chatbot:
*Planner is now a list of Task
*Add/Modify list, done, and chat methods.

Commit:
Add a reply for wrong 'done' task

Commit:
Modify runtest.sh to compile all the classes inside the java folder.
Remove all the javadocs as all the method are not finalised

Commit:
Remove duplicate 'by' and 'at' when printing Deadline and Event objects.

Commit:
Add DukeExceptions.java and all its sub-classes.
Add completeTask method and constants error phrases.
Add the try-catch block to chat, taskHandler and completeTask methods.

Commit:
Change completeTask method to taskAction method so that it will be able to handle 'delete'
Modify the DukeEmptyDoneActionException.java to handle 'delete'

Commit:
Edit Synthax

Commit:
Add Requirements for Level 7

Create a Storage class to manage finding/creating duke.txt file
Add DukeFileNotFoundException to throw when file is not found while loading the duke.txt
Add boolean to the constructor of Task class

Commit:
Add Requirements for Level 8

Convert the String property 'by' in Deadline and Event classes to LocalDate
Modify Deadline and Event toString method to use Dateformatter
Create DukeDateTimeParse.Exception.java

Commit:
Merge 'branch-Level-8' and 'branch-Level-7' to master

Modify chatbot to save even before  the "bye" input

* branch-Level-8:
  Add Requirements for Level 8

# Conflicts:
#	src/main/java/Chatbot.java
#	src/main/java/Deadline.java
#	src/main/java/Event.java

Commit:
Add Increments - A-MoreOOP

Add all the exception messages to their respective exception classes.
Transfer code of Chatbot to Duke.java and delete Chatbot.java
Create all Command classes which makes use of Storage, Ui and TaskList
Create Parse class to handle all the management of user input

Commit:
Replace LocalDate with LocalDateTime

Adjust the Parser, Exception, Storage, Deadline and Event class methods accordingly to match the format 'yyyy-mm-dd HH:mm'.

Commit:
Add JUnit tests

Modify exception messages.
Create ParserTest.java and TaskListTest.java

Commit:
Package the App as a JAR file

Commit:
Add JavaDoc comments

Delete enum TaskType

Commit:
Tweak the code to comply with a coding standard

Commit:
Give users a way to find a task by searching for a keyword.

Create FindCommand class.
Add methods to Ui, TaskList and Parses classes.

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc comments

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Tweak the code to comply with a coding standard

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/commands/Command.java

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Give users a way to find a task by searching for a keyword.

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java

Commit:
Modify tests for exceptions

Commit:
Merge commit '3b19ba1790d6035a9e246f5d4ae627bf68778606'

* commit '3b19ba1790d6035a9e246f5d4ae627bf68778606':
  Add Gradle support

Commit:
Fix Import of Gradle

Commit:
Fix test classes directory

Commit:
Use CheckStyle and Correct Checkstyle

Commit:
Config CheckStyle

Commit:
Set Up Java Fx

Commit:
Incorporate .fxml files

Commit:
Incorporate GUI with Duke

Commit:
Add Images

Commit:
Use var-args

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Use var-args
  Add Images
  Incorporate GUI with Duke
  Incorporate .fxml files
  Set Up Java Fx

Commit:
Merge branch 'branch-CheckStyle'

* branch-CheckStyle:
  Config CheckStyle
  Use CheckStyle and Correct Checkstyle

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/exceptions/DukeAlreadyDoneException.java
#	src/main/java/duke/exceptions/DukeDateTimeParseException.java
#	src/main/java/duke/exceptions/DukeDeadlineFormatException.java
#	src/main/java/duke/exceptions/DukeEmptyActionException.java
#	src/main/java/duke/exceptions/DukeEmptyDeadlineException.java
#	src/main/java/duke/exceptions/DukeEmptyEventException.java
#	src/main/java/duke/exceptions/DukeEmptyFindException.java
#	src/main/java/duke/exceptions/DukeEmptyToDoException.java
#	src/main/java/duke/exceptions/DukeEventFormatException.java
#	src/main/java/duke/exceptions/DukeFileNotFoundException.java
#	src/main/java/duke/exceptions/DukeInvalidIndexException.java
#	src/main/java/duke/exceptions/DukeInvalidTaskException.java

the-reefshark commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,78 @@
+        this.storage = new Storage();
+        this.ui = new Ui();
+        try {
+            this.tasks = new TaskList(storage.load());
```
Why did you use `this.tasks` instead of `tasks`?

the-reefshark commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,78 @@
+     * Executes the chatbot. The chatbot will respond
+     * based on specific keywords.
+     */
+    public void chat() {
```
Is the function body missing?

jimvae replied:
Thank you for pointing this out!

the-reefshark commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    /*
+    public static void main(String[] args) {
```
Should this be removed if not being used?

the-reefshark commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,282 @@
+     * DeleteCommand, DoneCommand and ExitCommand.
+     * @throws DukeException Throws exceptions that are all sub class of the DukeException.
+     */
+    public static Command parse(String commandLine) throws DukeException {
```
Great job with the styling of this method, looks clean

the-reefshark commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,282 @@
+     * @param string The user input.
+     * @return Returns a boolean.
+     */
+    static boolean isToDo(String string) {
```
Should "ToDo" be written as "Todo" instead like the next method?

the-reefshark commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,138 @@
+     * @return An ArrayList of tasks.
+     * @throws DukeFileNotFoundException Checks if the file is deleted or corrupted halfway.
+     */
+    ArrayList<Task> load() throws DukeFileNotFoundException {
```
Good use of comments and spacing to make this block of code readable

the-reefshark commented on src/test/java/TaskListTest.java:
```diff
@@ -0,0 +1,86 @@
+/**
+ * Tests the methods and exceptions for the TaskList class.
+ */
+public class TaskListTest {
```
Tests are very neatly structured and readable

the-reefshark reviewed (commented):
Well done with regards to styling and following coding practices, just some very minor areas to look into.

neoruien commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,78 @@
+}
+*/
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Should this method header comment start with Returns... or Gets... instead of the narrative sentence?

neoruien replied:
Should this method header comment start with "Gets..." or "Returns..." instead of this narrative sentence?

neoruien commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,282 @@
+    }
+
+    /**
+     * The main method that analyse the user input and give an
+     * appropriate command.
```
Should this method header comment start with "Analyses.." instead?

neoruien replied:
I noticed the same issue in several other places too.

jimvae replied:
Thank you for pointing this out!

neoruien commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,282 @@
+     * @throws DukeEventFormatException Checks if event contains a date and time.
+     * @throws DukeDateTimeParseException Checks if the data and time is correct.
+     */
+    private static Command parseAddCommandWithDate(String commandLine) throws DukeEmptyDeadlineException,
```
I like how you added comments to different parts of the method. It improves readability.

neoruien commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,282 @@
+        this.commandLine = commandLine;
+    }
+
+    /**
+     * The main method that analyse the user input and give an
+     * appropriate command.
```
Should this method header comment start with "Analyses..." instead?

neoruien replied:
I noticed the same issue in several other places too.

neoruien commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,282 @@
+     * @param commandLine This is part of the user input.
+     * @return A LocalDateTime object, based on the user input.
+     */
+    static LocalDateTime parseDateTime(String commandLine) {
```
I like how the individual components of the method body are attached with a comment. It improves readability.

neoruien commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,138 @@
+            for (Task task: tasks) {
+                if (task instanceof ToDo) {
+                    myWriter.write("T | "
+                            + (task.getIsDone() ? "1 | " : "0 | ")
+                            + task.getDescription());
+                } else if (task instanceof Deadline) {
+                    myWriter.write("D | "
+                            + (task.getIsDone() ? "1 | " : "0 | ")
+                            + task.getDescription() + " | "
+                            + ((Deadline) task).getDate().format(style));
+                } else {
+                    myWriter.write("E | "
+                            + (task.getIsDone() ? "1 | " : "0 | ")
+                            + task.getDescription() + " | "
+                            + ((Event) task).getDate().format(style));
```
The line breaks were added at appropriate locations. Good job!

neoruien commented:
LGTM. Just a few nits with the method header comments to fix. Good job!

Commit:
Fix code based from pull request comments

Update Java Docs

Make the anchorpane responsive

Commit:
Change mainClassName to Launcher

Commit:
Use Assertions

Commit:
Improve code quality

Commit:
Use Streams

Commit:
Use Lambdas

jimvae committed:
Merge pull request #1 from jimvae/branch-A-Assertions

Use Assertions

jimvae committed:
Merge pull request #2 from jimvae/branch-A-CodeQuality

 Improve code quality

jimvae committed:
Merge pull request #3 from jimvae/branch-A-Streams

Use Streams

jimvae committed:
Merge branch 'master' into branch-A-Lambdas

jimvae committed:
Merge pull request #4 from jimvae/branch-A-Lambdas

Use Lambdas

Commit:
Add an extension, B-Reminders

Create a TaskListFilter to easily filter the tasklist
Add reminder for deadlines/events this week when duke window is just opened

Commit:
Checkstyle check

Commit:
Improve the GUI

Commit:
Checkstyle

Commit:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Checkstyle
  Improve the GUI

Commit:
Add Ui.png

Commit:
Add a User Guide and screenshots

Commit:
Add a User Guide

Commit:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Add a User Guide
  Add a User Guide and screenshots

Commit:
Update README.md

jimvae committed:
Update README.md

Commit:
Add acknowledgments

jimvae committed:
Update README.md

jimvae committed:
Set theme jekyll-theme-cayman

jimvae committed:
Set theme jekyll-theme-cayman

jimvae committed:
Update README.md

jimvae committed:
Update README.md

jimvae committed:
Update README.md
